core/src/banking_stage/packet_deserializer.rs

```diff
@@ -198,6 +198,19 @@ impl PacketDeserializer {
             }
         })
     }
+
+    #[allow(dead_code)]
+    pub(crate) fn deserialize_packets_with_indexes(
+        packet_batch: &PacketBatch,
+    ) -> impl Iterator<Item = (ImmutableDeserializedPacket, usize)> + '_ {
+        let packet_indexes = PacketDeserializer::generate_packet_indexes(packet_batch);
+        packet_indexes.into_iter().filter_map(move |packet_index| {
+            let packet = packet_batch[packet_index].clone();
+            ImmutableDeserializedPacket::new(packet)
+                .ok()
+                .map(|packet| (packet, packet_index))
+        })
+    }
```

Comment:
just FYI this is next on item list to change after #4508 and #4518. At that point there's no point of `ImmutableDeserializedPacket` in `BankingStage`

Reply:
No issue with the code you posted, just letting you know depending on merge order we may see conflicts here, or the code will change shortly after this is merged

Reply:
thanks for the tip!